modular_skyrat/modules/colony_fabricator/code/construction/turfs.dm

```diff
@@ -0,0 +1,176 @@
+	desc = "A conservatively built metal frame with plastic paneling covering a thin air-seal layer. \
+		It's a little unnerving, but its better than nothing at all."
+	icon = 'modular_skyrat/modules/colony_fabricator/icons/prefab_wall.dmi'
+	icon_state = "prefab-0"
+	base_icon_state = "prefab"
+	can_engrave = FALSE
+	girder_type = null
+	hardness = 70
+	slicing_duration = 5 SECONDS
+	sheet_type = /obj/item/stack/sheet/plastic_wall_panel
+	sheet_amount = 1
+
+GLOBAL_LIST_INIT(plastic_wall_panel_recipes, list(
+	new/datum/stack_recipe("prefabricated wall", /turf/closed/wall/prefab_plastic, time = 3 SECONDS, one_per_turf = TRUE, on_solid_ground = TRUE, category = CAT_STRUCTURE), \
+	new/datum/stack_recipe("prefabricated window", /obj/structure/window/reinforced/colony_fabricator, time = 0.5 SECONDS, on_solid_ground = TRUE, check_direction = TRUE, is_fulltile = TRUE, category = CAT_WINDOWS), \
+	))
+
+/obj/item/stack/sheet/plastic_wall_panel
+	name = "plastic panels"
+	singular_name = "plastic panel"
+	desc = "What better material to make the walls of your soon to be home out of than sheets of flimsy plastic? \
+		Metal? What are you talking about, metal walls, in this economy? May also be used to make structures other \
+		than walls."
+	icon = 'modular_skyrat/modules/colony_fabricator/icons/tiles_item.dmi'
+	icon_state = "sheet-plastic"
+	inhand_icon_state = "sheet-plastic"
+	mats_per_unit = list(
+		/datum/material/plastic = HALF_SHEET_MATERIAL_AMOUNT,
+		/datum/material/glass = HALF_SHEET_MATERIAL_AMOUNT,
+	)
+	has_unique_girder = TRUE
+	material_type = /datum/material/plastic
+	merge_type = /obj/item/stack/sheet/plastic_wall_panel
+	walltype = /turf/closed/wall/prefab_plastic
+
+/obj/item/stack/sheet/plastic_wall_panel/examine(mob/user)
+	. = ..()
+	. += span_notice("You can build a prefabricated wall by right clicking on an empty floor.")
+
+/obj/item/stack/sheet/plastic_wall_panel/afterattack_secondary(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!isopenturf(target))
+		return SECONDARY_ATTACK_CONTINUE_CHAIN
+	var/turf/open/build_on = target
+	if(!user.Adjacent(build_on))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(isgroundlessturf(build_on))
+		user.balloon_alert(user, "can't place it here!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(build_on.is_blocked_turf())
+		user.balloon_alert(user, "something is blocking the tile!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(get_amount() < 1)
+		user.balloon_alert(user, "not enough material!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(!do_after(user, 3 SECONDS, build_on))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(build_on.is_blocked_turf())
+		user.balloon_alert(user, "something is blocking the tile!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(!use(1))
+		user.balloon_alert(user, "not enough material!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	new walltype(build_on)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
```

Comment:
I feel like all these cancels could be put into `.` instead, but I don't care enough to hold this PR back for this.